internal/dataplane/parser/translate_ingress.go

```diff
@@ -217,6 +218,25 @@ func ingressV1ToKongServiceLegacy(
 	return servicesCache
 }
 
+// routeName generates a name for a Kong Route based on the Ingress name, namespace and rule index and path index.
+func routeName(failuresCollector *failures.ResourceFailuresCollector, objIngress client.Object, ruleIndex, pathIndex int) *string {
+	// Indexes are expected to be ascending (starts from 0), hence the first conflict is expected to be
+	// for the below condition - both will be `111`. Register failure with a hint for user.
+	if ruleIndex == 1 && pathIndex == 11 || ruleIndex == 11 && pathIndex == 1 {
```

Comment:
We can hit this for multiple combinations, correct? Although unlikely, you could have a rule 2 with 22 paths and rule 22 with 2 paths (ignore the 0 index or add 1 to everything) and still hit the issue, even if your rule 1 and 11 only have 2 paths.

Should we instead check whether a particular `ruleIndex + pathIndex` string has already come up to handle even the more ridiculous rule/path counts?

Reply:
My rationale is that you can't hit `2` and `22` without hitting `1` and `11` before and having only `111` is not different from having beside `111` also `222` and `333` too, the configuration is wrong and a user will be notified. Am I missing something?

Reply:
If your rules with index 1 and 11 only have a single path each, they should only generate suffixes `1.0` and `11.0` if I'm thinking through it correctly. You can then independently have rules index 2 and 22 each have enough path to hit `2.22` and `22.2` (fake periods to make it a bit easier to read).

The other bit I forgot earlier is that the `ruleIndex == 1 && pathIndex == 11 || ruleIndex == 11 && pathIndex == 1` doesn't require that the other rule/path combo exists: it will be true if you have a rule 1 and path 11 even if there's no rule 11 and path 1 to conflict with it.